docs/fundamentals/networking/http/httpclient-migrate-from-httpwebrequest.md

```diff
@@ -0,0 +1,136 @@
+helpviewer_keywords: 
+  - "protocols, HTTP"
+  - "migration, HTTP"
+  - "HTTP"
+  - "HttpWebRequest class, about HttpWebRequest class"
+  - "application protocols, HTTP"
+  - "HttpClient class, about HttpClient class"
+  - "data requests, HTTP"
+  - "Internet, HTTP"
+---
+
+# HttpWebRequest to HttpClient Migration Guide
+
+This document aims to guide developers through the process of migrating from <xref:System.Net.HttpWebRequest>, <xref:System.Net.ServicePoint>, and <xref:System.Net.ServicePointManager> to <xref:System.Net.Http.HttpClient>. The migration is necessary due to the obsolescence of the older APIs and the numerous benefits offered by <xref:System.Net.Http.HttpClient>, including improved performance, better resource management, and a more modern and flexible API design. By following the steps outlined in this document, developers will be able to transition their codebases smoothly and take full advantage of the features provided by <xref:System.Net.Http.HttpClient>.
+
+## Migration steps
+
+Here is explanation for detailed migration steps.
+
+### Migrating from <xref:System.Net.HttpWebRequest> to <xref:System.Net.Http.HttpClient>
+
+The migration from <xref:System.Net.HttpWebRequest> to <xref:System.Net.Http.HttpClient> is essential due to the modern features and improved performance that <xref:System.Net.Http.HttpClient> offers. <xref:System.Net.Http.HttpClient> provides a more flexible and efficient way to make HTTP requests and handle responses. Here are the key steps and considerations for this migration:
+
+Let's start with some examples:
+
+#### Simple GET Request Using <xref:System.Net.HttpWebRequest>
+
+Here's an example of how the code might look:
+
+```c#
+using System.Net;
+
+HttpWebRequest request = WebRequest.CreateHttp(uri);
+using WebResponse response = await request.GetResponseAsync();
+```
+
+#### Simple GET Request Using <xref:System.Net.Http.HttpClient>
+
+Here's an example of how the code might look:
+
+```c#
+using System.Net.Http;
+
+using HttpClient client = new();
+using HttpResponseMessage message = await client.GetAsync(uri);
+```
+
+#### Simple POST Request Using <xref:System.Net.HttpWebRequest>
+
+Here's an example of how the code might look:
+
+```c#
+using System.Net;
+
+HttpWebRequest request = WebRequest.CreateHttp(uri);
+request.Method = "POST";
+request.ContentType = "text/plain";
+await using Stream stream = await request.GetRequestStreamAsync();
+await stream.WriteAsync("Hello World!"u8.ToArray());
+using WebResponse response = await request.GetResponseAsync();
+Memory<byte> buffer = new byte[1024];
+await using Stream responseStream = response.GetResponseStream();
+```
+
+#### Simple POST Request Using <xref:System.Net.Http.HttpClient>
+
+Here's an example of how the code might look:
+
+```c#
+using System.Net.Http;
+
+using HttpClient client = new();
+using HttpResponseMessage responseMessage = await client.PostAsync(uri, new StringContent("Hello World!"));
+```
+
+## 1:1 API Mapping Table
```

Comment:
it's not really 1:1 if some members don't have an equivalent API. Maybe something like this?

```suggestion
## Migrating ServicePointManager usage
```